[SC-10229] update github-discovery docs (#836)
RoadieHQ/marketing-site: content/docs/integrations/github-discovery/index.md

> 

Reviewer: If you add a `lastUpdated` key to the front matter YAML of this page, it will tell the user when the document was last updated, which is useful if they are trying to determine whether or not they should trust it.

## Introduction

> Roadie can load your catalog files automatically from GitHub. It takes advantage of the github webhooks to keep the catalog up to date with your additions and modifications.

Reviewer: ```suggestion
Roadie can load your catalog files automatically from GitHub. It takes advantage of GitHub Webhooks to keep the catalog up to date with your additions and modifications.
```

> 

Reviewer: If this is going to be enabled by default, then the goal of the document should be to teach the user how to change or add to the default settings.

I think we can do more in the intro to help the user understand what they will get out of reading this document. Why would I want to read it? What will I learn?

- You must be an Admin of Backstage

## Steps to add

1. Visit the Administration > Settings page and click on GitHub under Integrations
2. Scroll down to the `Automatically add and remove entities from your Catalog` section.
   ![add and remove](./default-settings.webp)
3. Press the `+ ADD ITEM` button in the bottom right corner.
4. In the input that appears enter a url that you'd like to have roadie to listen for additions and deletions.
   This url accepts regex in every segment
5. Press `SAVE`
6. Press `APPLY & RESTART` This will restart your instance it can take a couple of seconds.

## Steps to remove

1. Visit the Administration > Settings page and click on GitHub under Integrations
2. Scroll down to the `Automatically add and remove entities from your Catalog` section.
   ![add and remove](./default-settings.webp)
3. Press the minus sign next to the url.
4. Press `SAVE`
5. Press `APPLY & RESTART` This will restart your instance it can take a couple of seconds.

## Example configs

### Exact url

If you do not specify any wildcards it will load only the specified exact path.

```yaml
https://github.com/RoadieHQ/roadie-backstage-plugins/blob/main/catalog-info.yaml
```

> Roadie will load only catalog-info.yaml file from the main branch when it got added/removed from the repo roadie-backstage-plugins in the RoadieHQ organization

Reviewer 2: ```suggestion
In this example, Roadie will only load catalog-info.yaml files from the `main` branch when they get added/removed from the repo `roadie-backstage-plugins` in the `RoadieHQ` organization.
```

### Using the default branch

If you have default branch configured differently for your repos you can use the - sign as a substitution in your url and it will get replaced by your repo's default branch

```yaml
https://github.com/RoadieHQ/roadie-backstage-plugins/blob/-/catalog-info.yaml
```

> Roadie will load only catalog-info.yaml file from the default branch when it got added/removed from the repo roadie-backstage-plugins in the RoadieHQ organization

Reviewer 2: ```suggestion
In this example, Roadie will only load catalog-info.yaml files from the `main` branch when they get added/removed from the repo `roadie-backstage-plugins` in the `RoadieHQ` organization.
```

### Using wildcards

If you would like to catalog a file from all of your repos you can use the \* as a wildcard in any of the segmenets.

```yaml
https://github.com/RoadieHQ/*/blob/-/catalog-info.yaml
```

> Roadie will load only catalog-info.yaml file from the default branch when it got added/removed from any repo in the RoadieHQ organization

Reviewer 2: ```suggestion
In this example, Roadie will load a `catalog-info.yaml` file from the default branch when it gets added/removed from any repo in the RoadieHQ organization
```

### Putting your catalog files inside a folder

You can specify a folder in the resource part of the URL any depth you would like. This gives the possibility to specify the following url.

```yaml
https://github.com/RoadieHQ/*/blob/-/.roadie/*.yaml
```

> Roadie will load any yaml file inside a root level .roadie folder from the default branch when it got added/removed from any repo in the RoadieHQ organization

Reviewer 2: ```suggestion
In this example, Roadie will load any `.yaml` file inside a root level `.roadie` folder from the default branch when it gets added/removed from any repo in the RoadieHQ organization
```

## Confirming it worked

> Add a backstage component file to a repository which will match the configured discovery url. After commiting it to the correct branch it should be almost immedietly in your catalog.

Reviewer 2: ```suggestion
Add a backstage component file to a repository which matches the configured discovery url. After committing it to the correct branch, you should see it almost immediately in your catalog.
```

## Steps

1. Visit the Administration > Settings page and click on GitHub under Integrations
2. Scroll down to the `Automatically add and remove entities from your Catalog` section.
   ![add and remove](./add-remove.webp)
3. Press the `+ ADD ITEM` button in the bottom right corner.
4. In the input that appears enter a url that you'd like to have roadie to listen for additions and deletions.
   This url accepts regexes in every segment:

> - https://github.com/RoadieHQ/roadie-backstage-plugins/blob/-/catalog-info.yaml // Roadie will only add entities when a catalog-info.yaml file was added/removed to the default branch from the roadie-backstage-plugins repo in the RoadieHQ organiztion.

Reviewer: I suggest moving these out of "Steps" and into a new section called "Example configurations".

Then you can introduce the goal of each example config and use a code block to show the user what they would put into the input to replicate the example.
